spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/factory/AddResponseHeaderGatewayFilterFactory.java

```diff
@@ -16,23 +16,40 @@
 
 package org.springframework.cloud.gateway.filter.factory;
 
+import java.util.Arrays;
+import java.util.List;
+
 import reactor.core.publisher.Mono;
 
 import org.springframework.cloud.gateway.filter.GatewayFilter;
 import org.springframework.cloud.gateway.filter.GatewayFilterChain;
 import org.springframework.cloud.gateway.support.ServerWebExchangeUtils;
+import org.springframework.core.style.ToStringCreator;
 import org.springframework.http.HttpHeaders;
+import org.springframework.util.StringUtils;
 import org.springframework.web.server.ServerWebExchange;
 
 import static org.springframework.cloud.gateway.support.GatewayToStringStyler.filterToStringCreator;
 
 /**
  * @author Spencer Gibb
  */
-public class AddResponseHeaderGatewayFilterFactory extends AbstractNameValueGatewayFilterFactory {
+public class AddResponseHeaderGatewayFilterFactory
+		extends AbstractGatewayFilterFactory<AddResponseHeaderGatewayFilterFactory.Config> {
+
+	private static final String OVERRIDE_KEY = "override";
+
+	public AddResponseHeaderGatewayFilterFactory() {
+		super(Config.class);
+	}
 
 	@Override
-	public GatewayFilter apply(NameValueConfig config) {
```

Comment:
Would it be possible to leave this `apply` method signature and call the new one?  Trying to avoid breaking changes